includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1114,6 +1114,7 @@ public function process_payment_for_order( $cart, $payment_information, $schedul
 				$request->set_off_session( $payment_information->is_merchant_initiated() );
 				$request->set_payment_methods( $payment_methods );
 				$request->set_cvc_confirmation( $payment_information->get_cvc_confirmation() );
+				$request->set_shipping( $this->get_shipping_data_from_order( $order ) );
```

Comment:
Also, in relevant to my comments having shipping required for a BNPL, should we consider this being added conditionally for BNPLs only?

Reply:
Added a filter for sending this only for Afterpay where it is needed. 
[8a93f0f](https://github.com/Automattic/woocommerce-payments/pull/6492/commits/8a93f0f43e99a07ce6f4b986d67e8e20c1883907)

Reply:
Here too for the JS bit
17d2033